mitodl/ocw-to-hugo
# move other_versions to course.json

#### Pre-Flight checklist


#### What are the relevant tickets?
Closes https://github.com/mitodl/ocw-to-hugo/issues/291

#### What's this PR do?
In https://github.com/mitodl/ocw-to-hugo/pull/283, we created a reverse lookup to find other versions of a course using the `master_subject` system from the OCW legacy data.  This data was set up to be deposited into the course home page's front matter.  This PR moves that data into `course.json`, the course data template, for easier access and ingestion.

#### How should this be manually tested?
 - Read the readme and make sure you are set up to download courses from S3
 - Put the following in `private/courses.json`:
```
{
    "courses": [
        "8-01sc-classical-mechanics-fall-2016",
        "8-01l-physics-i-classical-mechanics-fall-2005",
        "8-01x-physics-i-classical-mechanics-with-an-experimental-focus-fall-2002"
    ]
}
```
 - Convert the courses using: `node . -i private/input -o private/output --courses private/courses.json --download --rm`
 - Inspect the contents of `private/output/8-01sc-classical-mechanics-fall-2016/data/course.json` and ensure that the alternate versions are in the data template:
```
  "other_versions": [
    "[8.01L PHYSICS I: CLASSICAL MECHANICS](/courses/8-01l-physics-i-classical-mechanics-fall-2005) |  FALL 2005",
    "[8.01X PHYSICS I: CLASSICAL MECHANICS WITH AN EXPERIMENTAL FOCUS](/courses/8-01x-physics-i-classical-mechanics-with-an-experimental-focus-fall-2002) |  FALL 2002"
  ]
```
